python/ql/src/semmle/python/Concepts.qll

```diff
@@ -314,7 +314,7 @@ module HTTP {
       string getUrlPattern() { result = range.getUrlPattern() }
 
       /** Gets a function that will handle incoming requests for this route, if any. */
-      Function getARouteHandler() { result = range.getARouteHandler() }
+      Function getARequestHandler() { result = range.getARequestHandler() }
```

Comment:
Could this return a `RequestHandler`? (likewise in `Range`)

Reply:
This was something I struggled a bit with myself. Thanks for pointing it out, I should definitely have added a comment about this myself.

With the current setup, since the result of `getARouteHandler`  is a `Function`. This means if you're modeling a web-library that uses "route to specific function/class" pattern (like flask/Django/Tornado), it's easy to specify the request handler function: [Example of defining a route-setup in flask](https://github.com/github/codeql/blob/b794fcb841badee807c082817b061fc86ed70e0e/python/ql/src/semmle/python/frameworks/Flask.qll#L290-L305).

If we changed the result of `getARouteHandler` to be a `HTTP::Server::RequestHandler`, then we need to write some boiler code to make things work. Why? Since `RequestHandler` doesn't include any functions by default, we need a separate class to extend `HTTP::Server::RequestHandler::Range` before `getARouteHandler` can have a valid result.

Writing this up, it seems like I was faced with a trade-off between making it easy to _write_ library models, and _using_ library models, and I chose to make it easy to _write_ them. That's probably the wrong trade-off. Can I fix this in a separate PR? I know deep down in my engineering heart that I should probably do it in this PR since this is the PR that introduces the concept, but I also know that it'll probably take me a bit of time, and I also have an urge to get this (and the Tornado PR) merged :smile: Let me know what you think :wink:

Reply:
Doing it in a separate PR is probably fine, especially if it will touch many separate places. That PR will just need some explanatory text to not be confusing. If that is now the intention, we might add a comment already now to that effect.

Reply:
sure, added a comment now :blush: